sdk/test/logs/logger_sdk_test.cc

```diff
@@ -303,3 +309,46 @@ TEST(LoggerSDK, EventLog)
   ASSERT_EQ(shared_recordable->GetEventName(), "otel-cpp.event_name");
   ASSERT_EQ(shared_recordable->GetEventDomain(), "otel-cpp.event_domain");
 }
+
+// Define a basic Logger class
+class TestLogger : public opentelemetry::logs::Logger
+{
+  const nostd::string_view GetName() noexcept override { return "test logger"; }
+
+  nostd::unique_ptr<opentelemetry::logs::LogRecord> CreateLogRecord() noexcept override
+  {
+    return nullptr;
+  }
+
+  using Logger::EmitLogRecord;
+
+  void EmitLogRecord(nostd::unique_ptr<opentelemetry::logs::LogRecord> &&) noexcept override {}
+};
+
+// Define a basic LoggerProvider class that returns an instance of the logger class defined above
+class TestProvider : public opentelemetry::logs::LoggerProvider
+{
+  nostd::shared_ptr<opentelemetry::logs::Logger> GetLogger(
+      nostd::string_view /* logger_name */,
+      nostd::string_view /* library_name */,
+      nostd::string_view /* library_version */,
+      nostd::string_view /* schema_url */,
+      const opentelemetry::common::KeyValueIterable & /* attributes */) override
+  {
+    return nostd::shared_ptr<opentelemetry::logs::Logger>(new TestLogger());
+  }
+};
+
+TEST(Logger, PushLoggerImplementation)
+{
+  // Push the new loggerprovider class into the global singleton
+  auto test_provider = nostd::shared_ptr<opentelemetry::logs::LoggerProvider>(new TestProvider());
+  logs_sdk::Provider::SetLoggerProvider(test_provider);
+
+  auto lp = logs_api::Provider::GetLoggerProvider();
+
+  // Check that the implementation was pushed by calling TestLogger's GetName()
+  nostd::string_view schema_url{"https://opentelemetry.io/schemas/1.11.0"};
+  auto logger = lp->GetLogger("TestLogger", "opentelelemtry_library", "", schema_url);
+  ASSERT_EQ("test logger", logger->GetName());
+}
```

Comment:
Not sure if I am missing something, is this test related to getting logger/provider with OTEL_SDK_ENABLED?

Reply:
This test was in the API, in logger_test.cc, and was moved to the SDK when the setter was no longer in the API.

Now that the changes in API are reverted, this test is no longer needed in the SDK, I will clean this up.

Good catch.